Please support euclid 0.20

I tried to build `lyon_geom` against `euclid` 0.20, and I’m getting a lot of errors due to API changes:
```
error[E0107]: wrong number of type arguments: expected 2, found 1
  --> src/quadratic_bezier.rs:23:15
   |
23 |     pub from: Point<S>,
   |               ^^^^^^^^ expected 2 type arguments
```

```
error[E0107]: wrong number of type arguments: expected 3, found 1
   --> src/quadratic_bezier.rs:248:41
    |
248 |     pub fn transform(&self, transform: &Transform2D<S>) -> Self {
    |                                         ^^^^^^^^^^^^^^ expected 3 type arguments
```
```
error[E0107]: wrong number of type arguments: expected 3, found 1
   --> src/lib.rs:253:27
    |
253 |     pub type Rotation2D = euclid::Rotation2D<f32>;
    |                           ^^^^^^^^^^^^^^^^^^^^^^^ expected 3 type arguments

error: aborting due to 167 previous errors
```
Could you please have a look and update to the new API?

Thanks!

--- a/algorithms/src/aabb.rs
+++ b/algorithms/src/aabb.rs
@@ -137,13 +137,13 @@ impl TightBoundingRect for PathEvent {
             }
             PathEvent::Quadratic(ref segment) => {
                 let r = segment.bounding_rect();
-                *min = Point::min(*min, r.origin);
-                *max = Point::max(*max, r.bottom_right());
+                *min = Point::min(*min, r.min());
+                *max = Point::max(*max, r.max());
             }
             PathEvent::Cubic(ref segment) => {
                 let r = segment.bounding_rect();
-                *min = Point::min(*min, r.origin);
-                *max = Point::max(*max, r.bottom_right());
+                *min = Point::min(*min, r.min());
+                *max = Point::max(*max, r.max());
             }
             PathEvent::Close(..) => {}
         }
@@ -163,8 +163,8 @@ impl TightBoundingRect for QuadraticEvent {
             }
             QuadraticEvent::Quadratic(ref segment) => {
                 let r = segment.bounding_rect();
-                *min = Point::min(*min, r.origin);
-                *max = Point::max(*max, r.bottom_right());
+                *min = Point::min(*min, r.min());
+                *max = Point::max(*max, r.max());
             }
             QuadraticEvent::Close(..) => {}
         }

--- a/algorithms/src/advanced_path.rs
+++ b/algorithms/src/advanced_path.rs
@@ -83,12 +83,14 @@ impl AdvancedPath {
 
     /// Add a rectangular sub-path.
     pub fn add_rectangle(&mut self, rectangle: &Rect) -> SubPathId {
+        let min = rectangle.min();
+        let max = rectangle.min();
         self.add_polyline(
             &[
-                rectangle.origin,
-                rectangle.top_right(),
-                rectangle.bottom_right(),
-                rectangle.bottom_left(),
+                min,
+                point(max.x, min.y),
+                max,
+                point(min.x, max.y),
             ],
             true
         )

--- a/algorithms/src/fit.rs
+++ b/algorithms/src/fit.rs
@@ -42,8 +42,8 @@ pub fn fit_rectangle(src_rect: &Rect, dst_rect: &Rect, style: FitStyle) -> Trans
         FitStyle::Vertical => vector(scale.y, scale.y),
     };
 
-    let src_center = src_rect.origin.lerp(src_rect.bottom_right(), 0.5);
-    let dst_center = dst_rect.origin.lerp(dst_rect.bottom_right(), 0.5);
+    let src_center = src_rect.origin.lerp(src_rect.max(), 0.5);
+    let dst_center = dst_rect.origin.lerp(dst_rect.max(), 0.5);
 
     Transform2D::create_translation(-src_center.x, -src_center.y)
         .post_scale(scale.x, scale.y)
@@ -67,7 +67,7 @@ pub fn fit_path(path: &Path, output_rect: &Rect, style: FitStyle) -> Path {
 fn simple_fit() {
     fn approx_eq(a: &Rect, b: &Rect) -> bool {
         use crate::geom::euclid::approxeq::ApproxEq;
-        let result = a.origin.approx_eq(&b.origin) && a.bottom_right().approx_eq(&b.bottom_right());
+        let result = a.origin.approx_eq(&b.origin) && a.max().approx_eq(&b.max());
         if !result {
             println!("{:?} == {:?}", a, b);
         }

--- a/algorithms/src/hatching.rs
+++ b/algorithms/src/hatching.rs
@@ -29,7 +29,7 @@ use crate::path::PathEvent;
 use crate::path::builder::{Build, FlatPathBuilder, PathBuilder};
 use crate::geom::LineSegment;
 use crate::geom::math::{Point, Vector, point, vector};
-use crate::geom::euclid::{Angle, Rotation2D};
+use crate::geom::euclid::{Angle, default::Rotation2D};
 use std::marker::PhantomData;
 
 use std::cmp::Ordering;
@@ -330,7 +330,7 @@ impl Hatcher {
     ) {
         self.transform = Rotation2D::new(-options.angle);
         self.uv_origin = Rotation2D::new(options.angle).transform_point(
-            &options.uv_origin
+            options.uv_origin
         );
         self.active_edges.clear();
         self.segment.row = 0;
@@ -413,12 +413,12 @@ impl Hatcher {
             }
             let x = active_edge.solve_x_for_y(y);
             if self.compute_tangents {
-                tangent = self.transform.transform_vector(&active_edge.to_vector()).normalize();
+                tangent = self.transform.transform_vector(active_edge.to_vector()).normalize();
             }
 
             if inside {
-                self.segment.a.position = self.transform.transform_point(&point(prev_x, y));
-                self.segment.b.position = self.transform.transform_point(&point(x, y));
+                self.segment.a.position = self.transform.transform_point(point(prev_x, y));
+                self.segment.b.position = self.transform.transform_point(point(x, y));
                 self.segment.a.u = prev_x - self.uv_origin.x;
                 self.segment.b.u = x - self.uv_origin.x;
                 if self.compute_tangents {
@@ -471,8 +471,8 @@ impl EventsBuilder {
 
     fn add_edge(&mut self, from: Point, to: Point) {
         let rotation = Rotation2D::new(self.angle);
-        let mut from = rotation.transform_point(&from);
-        let mut to = rotation.transform_point(&to);
+        let mut from = rotation.transform_point(from);
+        let mut to = rotation.transform_point(to);
         if compare_positions(from, to) == Ordering::Greater {
             mem::swap(&mut from, &mut to);
         }

--- a/cli/src/show.rs
+++ b/cli/src/show.rs
@@ -201,7 +201,7 @@ pub fn show_path(cmd: TessellateCmd, render_options: RenderCmd) {
     let constants = factory.create_constant_buffer(1);
 
     let aabb = bounding_rect(cmd.path.iter());
-    let center = aabb.origin.lerp(aabb.bottom_right(), 0.5).to_vector();
+    let center = aabb.origin.lerp(aabb.max(), 0.5).to_vector();
 
     let mut scene = SceneParams {
         target_zoom: 1.0,

--- a/extra/src/triangle_rasterizer.rs
+++ b/extra/src/triangle_rasterizer.rs
@@ -5,7 +5,7 @@ use math::*;
 use euclid;
 use image::MutableImageSlice;
 
-type IntVector = euclid::Vector2D<i32>;
+type IntVector = euclid::default::Vector2D<i32>;
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 struct IntVec4 {

--- a/geom/Cargo.toml
+++ b/geom/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "lyon_geom"
-version = "0.12.7"
+version = "0.14.0"
 description = "2D quadratic and cubic bézier arcs and line segment math on top of euclid."
 authors = ["Nicolas Silva <nical@fastmail.com>"]
 repository = "https://github.com/nical/lyon"
@@ -17,7 +17,7 @@ name = "lyon_geom"
 serialization = ["serde", "euclid/serde"]
 
 [dependencies]
-euclid = "0.19.0"
+euclid = "0.20.0"
 arrayvec = "0.4"
 num-traits = "0.2"
 serde = {version = "1.0", optional = true, features = ["serde_derive"] }

--- a/geom/src/arc.rs
+++ b/geom/src/arc.rs
@@ -428,7 +428,7 @@ impl<S: Scalar> Arc<S> {
     fn tangent_at_angle(&self, angle: Angle<S>) -> Vector<S> {
         let a = angle.get();
         Rotation2D::new(self.x_rotation).transform_vector(
-            &vector(-self.radii.x * Float::sin(a), self.radii.y * Float::cos(a))
+            vector(-self.radii.x * Float::sin(a), self.radii.y * Float::cos(a))
         )
     }
 }
@@ -582,7 +582,7 @@ where
 
 fn sample_ellipse<S: Scalar>(radii: Vector<S>, x_rotation: Angle<S>, angle: Angle<S>) -> Point<S> {
     Rotation2D::new(x_rotation).transform_point(
-        &point(radii.x * Float::cos(angle.get()), radii.y * Float::sin(angle.get()))
+        point(radii.x * Float::cos(angle.get()), radii.y * Float::sin(angle.get()))
     )
 }
 

--- a/geom/src/cubic_bezier.rs
+++ b/geom/src/cubic_bezier.rs
@@ -292,10 +292,10 @@ impl<S: Scalar> CubicBezierSegment<S> {
     #[inline]
     pub fn transform(&self, transform: &Transform2D<S>) -> Self {
         CubicBezierSegment {
-            from: transform.transform_point(&self.from),
-            ctrl1: transform.transform_point(&self.ctrl1),
-            ctrl2: transform.transform_point(&self.ctrl2),
-            to: transform.transform_point(&self.to)
+            from: transform.transform_point(self.from),
+            ctrl1: transform.transform_point(self.ctrl1),
+            ctrl2: transform.transform_point(self.ctrl2),
+            to: transform.transform_point(self.to)
         }
     }
 

--- a/geom/src/lib.rs
+++ b/geom/src/lib.rs
@@ -191,23 +191,23 @@ mod scalar {
 }
 
 mod generic_math {
-    /// Alias for `euclid::Point2D`.
-    pub use euclid::Point2D as Point;
+    /// Alias for `euclid::default::Point2D`.
+    pub use euclid::default::Point2D as Point;
 
-    /// Alias for `euclid::Vector2D`.
-    pub use euclid::Vector2D as Vector;
+    /// Alias for `euclid::default::Vector2D`.
+    pub use euclid::default::Vector2D as Vector;
 
-    /// Alias for `euclid::Size2D`.
-    pub use euclid::Size2D as Size;
+    /// Alias for `euclid::default::Size2D`.
+    pub use euclid::default::Size2D as Size;
 
-    /// Alias for `euclid::Rect`
-    pub use euclid::Rect;
+    /// Alias for `euclid::default::Rect`
+    pub use euclid::default::Rect;
 
-    /// Alias for `euclid::Transform2D`
-    pub use euclid::Transform2D;
+    /// Alias for `euclid::default::Transform2D`
+    pub use euclid::default::Transform2D;
 
-    /// Alias for `euclid::Rotation2D`
-    pub use euclid::Rotation2D;
+    /// Alias for `euclid::default::Rotation2D`
+    pub use euclid::default::Rotation2D;
 
     /// An angle in radians.
     pub use euclid::Angle;
@@ -231,26 +231,26 @@ pub mod math {
 
     use euclid;
 
-    /// Alias for ```euclid::Point2D<f32>```.
-    pub type Point = euclid::Point2D<f32>;
+    /// Alias for ```euclid::default::Point2D<f32>```.
+    pub type Point = euclid::default::Point2D<f32>;
 
-    /// Alias for ```euclid::Point2D<f64>```.
-    pub type F64Point = euclid::Point2D<f64>;
+    /// Alias for ```euclid::default::Point2D<f64>```.
+    pub type F64Point = euclid::default::Point2D<f64>;
 
-    /// Alias for ```euclid::Point2D<f32>```.
-    pub type Vector = euclid::Vector2D<f32>;
+    /// Alias for ```euclid::default::Point2D<f32>```.
+    pub type Vector = euclid::default::Vector2D<f32>;
 
-    /// Alias for ```euclid::Size2D<f32>```.
-    pub type Size = euclid::Size2D<f32>;
+    /// Alias for ```euclid::default::Size2D<f32>```.
+    pub type Size = euclid::default::Size2D<f32>;
 
-    /// Alias for ```euclid::Rect<f32>```
-    pub type Rect = euclid::Rect<f32>;
+    /// Alias for ```euclid::default::Rect<f32>```
+    pub type Rect = euclid::default::Rect<f32>;
 
-    /// Alias for ```euclid::Transform2D<f32>```
-    pub type Transform2D = euclid::Transform2D<f32>;
+    /// Alias for ```euclid::default::Transform2D<f32>```
+    pub type Transform2D = euclid::default::Transform2D<f32>;
 
-    /// Alias for ```euclid::Rotation2D<f32>```
-    pub type Rotation2D = euclid::Rotation2D<f32>;
+    /// Alias for ```euclid::default::Rotation2D<f32>```
+    pub type Rotation2D = euclid::default::Rotation2D<f32>;
 
     /// An angle in radians (f32).
     pub type Angle = euclid::Angle<f32>;

--- a/geom/src/line.rs
+++ b/geom/src/line.rs
@@ -170,8 +170,8 @@ impl<S: Scalar> LineSegment<S> {
     #[inline]
     pub fn transform(&self, transform: &Transform2D<S>) -> Self {
         LineSegment {
-            from: transform.transform_point(&self.from),
-            to: transform.transform_point(&self.to),
+            from: transform.transform_point(self.from),
+            to: transform.transform_point(self.to),
         }
     }
 

--- a/geom/src/quadratic_bezier.rs
+++ b/geom/src/quadratic_bezier.rs
@@ -247,9 +247,9 @@ impl<S: Scalar> QuadraticBezierSegment<S> {
     #[inline]
     pub fn transform(&self, transform: &Transform2D<S>) -> Self {
         QuadraticBezierSegment {
-            from: transform.transform_point(&self.from),
-            ctrl: transform.transform_point(&self.ctrl),
-            to: transform.transform_point(&self.to)
+            from: transform.transform_point(self.from),
+            ctrl: transform.transform_point(self.ctrl),
+            to: transform.transform_point(self.to)
         }
     }
 

--- a/geom/src/triangle.rs
+++ b/geom/src/triangle.rs
@@ -76,9 +76,9 @@ impl<S: Scalar> Triangle<S> {
     #[inline]
     pub fn transform(&self, transform: &Transform2D<S>) -> Self {
         Triangle {
-            a: transform.transform_point(&self.a),
-            b: transform.transform_point(&self.b),
-            c: transform.transform_point(&self.c)
+            a: transform.transform_point(self.a),
+            b: transform.transform_point(self.b),
+            c: transform.transform_point(self.c)
         }
     }
 

--- a/path/Cargo.toml
+++ b/path/Cargo.toml
@@ -16,5 +16,5 @@ name = "lyon_path"
 serialization = ["serde", "lyon_geom/serialization"]
 
 [dependencies]
-lyon_geom = { version = "0.12.7", path = "../geom" }
+lyon_geom = { version = "0.14.0", path = "../geom" }
 serde = { version = "1.0", optional = true, features = ["serde_derive"] }

--- a/path/src/events.rs
+++ b/path/src/events.rs
@@ -118,8 +118,8 @@ impl QuadraticEvent {
 impl Transform for FlattenedEvent {
     fn transform(&self, mat: &Transform2D) -> Self {
         match self {
-            FlattenedEvent::MoveTo(ref to) => {
-                FlattenedEvent::MoveTo(mat.transform_point(to))
+            FlattenedEvent::MoveTo(to) => {
+                FlattenedEvent::MoveTo(mat.transform_point(*to))
             }
             FlattenedEvent::Line(ref segment) => {
                 FlattenedEvent::Line(segment.transform(mat))
@@ -134,8 +134,8 @@ impl Transform for FlattenedEvent {
 impl Transform for QuadraticEvent {
     fn transform(&self, mat: &Transform2D) -> Self {
         match self {
-            QuadraticEvent::MoveTo(ref to) => {
-                QuadraticEvent::MoveTo(mat.transform_point(to))
+            QuadraticEvent::MoveTo(to) => {
+                QuadraticEvent::MoveTo(mat.transform_point(*to))
             }
             QuadraticEvent::Line(ref segment) => {
                 QuadraticEvent::Line(segment.transform(mat))
@@ -153,7 +153,7 @@ impl Transform for QuadraticEvent {
 impl Transform for PathEvent {
     fn transform(&self, mat: &Transform2D) -> Self {
         match self {
-            PathEvent::MoveTo(ref to) => { PathEvent::MoveTo(mat.transform_point(to)) }
+            PathEvent::MoveTo(to) => { PathEvent::MoveTo(mat.transform_point(*to)) }
             PathEvent::Line(ref segment) => { PathEvent::Line(segment.transform(mat)) }
             PathEvent::Quadratic(ref segment) => { PathEvent::Quadratic(segment.transform(mat)) }
             PathEvent::Cubic(ref segment) => { PathEvent::Cubic(segment.transform(mat)) }

--- a/tessellation/src/basic_shapes.rs
+++ b/tessellation/src/basic_shapes.rs
@@ -35,6 +35,18 @@ use crate::{FillTessellator, TessellationResult};
 
 use std::f32::consts::PI;
 
+fn bottom_left(rect: &Rect) -> Point {
+    point(rect.min_x(), rect.max_y())
+}
+
+fn top_right(rect: &Rect) -> Point {
+    point(rect.max_x(), rect.min_y())
+}
+
+fn bottom_right(rect: &Rect) -> Point {
+    rect.max()
+}
+
 /// Tessellate a triangle.
 pub fn fill_triangle(
     v1: Point,
@@ -171,19 +183,19 @@ pub fn fill_rectangle(
     )?;
     let b = output.add_vertex(
         FillVertex {
-            position: rect.bottom_left(),
+            position: bottom_left(&rect),
             normal: vector(-1.0, 1.0),
         }
     )?;
     let c = output.add_vertex(
         FillVertex {
-            position: rect.bottom_right(),
+            position: bottom_right(&rect),
             normal: vector(1.0, 1.0),
         }
     )?;
     let d = output.add_vertex(
         FillVertex {
-            position: rect.top_right(),
+            position: top_right(&rect),
             normal: vector(1.0, -1.0),
         }
     )?;
@@ -206,9 +218,9 @@ pub fn stroke_rectangle(
 
     stroke_quad(
         rect.origin,
-        rect.top_right(),
-        rect.bottom_right(),
-        rect.bottom_left(),
+        top_right(&rect),
+        bottom_right(&rect),
+        bottom_left(&rect),
         options,
         output
     )
@@ -242,7 +254,7 @@ fn stroke_thin_rectangle(
     )?;
     let b = output.add_vertex(
         StrokeVertex {
-            position: rect.bottom_left(),
+            position: bottom_left(&rect),
             normal: vector(-1.0, 1.0),
             advancement: 0.0,
             side: Side::Left,
@@ -250,7 +262,7 @@ fn stroke_thin_rectangle(
     )?;
     let c = output.add_vertex(
         StrokeVertex {
-            position: rect.bottom_right(),
+            position: bottom_right(&rect),
             normal: vector(1.0, 1.0),
             advancement: 1.0,
             side: Side::Right,
@@ -258,7 +270,7 @@ fn stroke_thin_rectangle(
     )?;
     let d = output.add_vertex(
         StrokeVertex {
-            position: rect.top_right(),
+            position: top_right(&rect),
             normal: vector(1.0, -1.0),
             advancement: 1.0,
             side: Side::Right,

--- a/tessellation/src/geometry_builder.rs
+++ b/tessellation/src/geometry_builder.rs
@@ -191,7 +191,7 @@
 //! ```
 //! use lyon_tessellation::geometry_builder::*;
 //! use lyon_tessellation::{FillVertex, TessellationResult};
-//! use lyon_tessellation::math::{Rect, vector};
+//! use lyon_tessellation::math::{Rect, vector, point};
 //!
 //! // A tessellator that generates an axis-aligned quad.
 //! // Returns a structure containing the number of vertices and number of indices allocated
@@ -202,17 +202,19 @@
 //! {
 //!     output.begin_geometry();
 //!     // Create the vertices...
+//!     let min = rect.min();
+//!     let max = rect.min();
 //!     let a = output.add_vertex(
-//!         FillVertex { position: rect.origin, normal: vector(-1.0, -1.0) }
+//!         FillVertex { position: min, normal: vector(-1.0, -1.0) }
 //!     )?;
 //!     let b = output.add_vertex(
-//!         FillVertex { position: rect.top_right(), normal: vector(1.0, -1.0) }
+//!         FillVertex { position: point(max.x, min.y), normal: vector(1.0, -1.0) }
 //!     )?;
 //!     let c = output.add_vertex(
-//!         FillVertex { position: rect.bottom_right(), normal: vector(1.0, 1.0) }
+//!         FillVertex { position: max, normal: vector(1.0, 1.0) }
 //!     )?;
 //!     let d = output.add_vertex(
-//!         FillVertex { position: rect.bottom_left(), normal: vector(-1.0, 1.0) }
+//!         FillVertex { position: point(min.x, max.y), normal: vector(-1.0, 1.0) }
 //!     )?;
 //!     // ...and create triangle form these points. a, b, c, and d are relative offsets in the
 //!     // vertex buffer.

--- a/tessellation/src/math_utils.rs
+++ b/tessellation/src/math_utils.rs
@@ -8,9 +8,9 @@ use std::f64;
 
 pub type FixedPoint32 = fixed::Fp32<fixed::_16>;
 pub type FixedPoint64 = fixed::Fp64<fixed::_16>;
-pub type TessVector = euclid::Vector2D<FixedPoint32>;
-pub type TessPoint = euclid::Point2D<FixedPoint32>;
-pub type TessPoint64 = euclid::Point2D<FixedPoint64>;
+pub type TessVector = euclid::default::Vector2D<FixedPoint32>;
+pub type TessPoint = euclid::default::Point2D<FixedPoint32>;
+pub type TessPoint64 = euclid::default::Point2D<FixedPoint64>;
 #[inline]
 pub fn fixed(val: f32) -> FixedPoint32 { FixedPoint32::from_f32(val) }
 

--- a/tessellation/src/path_fill.rs
+++ b/tessellation/src/path_fill.rs
@@ -1498,7 +1498,7 @@ impl ActiveEdge {
 // If two points have the same y coordinate, the one on the right (x pointing to the right)
 // is the one after.
 #[inline]
-fn is_after<T: PartialOrd>(a: euclid::Point2D<T>, b: euclid::Point2D<T>) -> bool {
+fn is_after<T: PartialOrd>(a: euclid::default::Point2D<T>, b: euclid::default::Point2D<T>) -> bool {
     a.y > b.y || (a.y == b.y && a.x > b.x)
 }
 

